package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -343,15 +357,15 @@ def _write_next_frame(self, obj):
            Use AtomGroup or Universe as an input instead.
         """
         if hasattr(obj, "atoms"):
-            if hasattr(obj, 'universe'):
+            if hasattr(obj, "universe"):
                 # For AtomGroup and children (Residue, ResidueGroup, Segment)
                 ts_full = obj.universe.trajectory.ts
                 if ts_full.n_atoms == obj.atoms.n_atoms:
                     ts = ts_full
                 else:
                     # Only populate a time step with the selected atoms.
-                    ts = ts_full.copy_slice(atoms.indices)
-            elif hasattr(obj, 'trajectory'):
+                    ts = ts_full.copy_slice(obj.atoms.indices)
```

Comment:
Can we add a test to cover this?

Reply:
I'm not sure how this code would be triggered from the high level API. I think this was copy/pasted from the XYZ writer, I can change the code to use the same one as the current version of the XYZ writer, and take inspiration from there.

Reply:
It looks like a case where you pass an atomgroup which only represents a portion of your full system, but I'm not 100% sure.

Either way, I fully support any changes that a) improves line coverage, b) makes it simpler for us to work out what the code is doing 🚀 

If you think you prefer addressing it in a separate PR that's also an option.